src/Fixer/ClassNotation/FinalInternalClassFixer.php

```diff
@@ -185,10 +199,6 @@ protected function createConfigurationDefinition(): FixerConfigurationResolverIn
      */
     private function isClassCandidate(Tokens $tokens, int $index): bool
     {
-        if ($tokens[$tokens->getPrevMeaningfulToken($index)]->isGivenKind([T_ABSTRACT, T_FINAL])) {
-            return false; // ignore class; it is abstract or already final
-        }
-
```

Comment:
not sure I follow, these lines have been removed in favor of `$modifiers = $tokensAnalyzer->getClassyModifiers($index);` ?

Reply:
Yes, I see it. The part before `but` was only my loud thinking (trying to understand the root cause of this change), so let's focus on what's after `but` (now in bold).

Behaviour of `isClassCandidate()` was changed here, some part of responsibility was extracted and is used directly in `applyFix()`. This is wrong IMHO. Support for `readonly` should be added to `isClassCandidate()` because it fits that method well - it's only new scenario for considering class as candidate (2 modifiers).

Is that clear now? 🙂

Reply:
:+1:   I've moved all checks into `isClassCandidate` method, let me know what you think

Reply:
It's OK now, I would only consider standardising approach for token analyzer - here you pass it as `isClassCandidate ()`' s argument, but in `src/Fixer/ClassNotation/ProtectedToPrivateFixer.php` it's used as private property (and I [pointed it in my first comment](#discussion_r973328679) 😉 ).

Reply:
I think that fits in the broader discussion of how the rules/fixers obtain things like analyzers, normalizers, utils and helpers etc., Maybe we want to move to a DI kind of solution so the factoring and wiring of these are not done by the individual rules. Anyway, i think this solution is OK for now as I think both ways are used already in the code base.